versions/1.0.0.md

```diff
@@ -480,6 +487,91 @@ criteria:
     - $statusCode == 503
 ```
 
+#### Component Object
+
+Holds a set of reusable objects for different aspects of the Workflows Specification. All objects defined within the components object will have no effect on the workflow unless they are explicitly referenced from properties outside the components object.
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---|---
+<a name="componentInputs"></a> inputs | Map[`string`, [Schema Object](#schemaObject)] | An object to hold reusable JSON Schema objects to be referenced from workflow inputs.
+<a name="componentParameters">parameters</a> | Map[string, [Parameter Object](#parameter-object)] | An object to hold reusable Parameter Objects
+
+This object MAY NOT be extended with [Specification Extensions](#specificationExtensions).
+
+All the fixed fields declared above are objects that MUST use keys that match the regular expression: `^[a-zA-Z0-9\.\-_]+$`. The key is used to refer to the input or parameter in other parts of the Workflow document.
+
+Field Name Examples:
+
+```
+User
+User_1
+User_Name
+user-name
+my.org.User
+```
+
+##### Component Object Example
+
+```json
+"components": {
+  "inputs": {
+    "storeId": {
+      "name": "storeId",
+      "in": "header",
+      "value": "$inputs.x-store-id"
+    }
+  },
+  "parameters": {
+    "pagination": {
+      "type": "object",
+      "properties": {
+        "page": {
+          "type": "integer",
+          "format": "int32"
+        },
+        "pageSize": {
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    }
+  }
+}
+```
+
+#### Reference Object
+
+**Note -** This section is derived from the [OpenAPI Specification, v3.1](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#referenceObject), and is intended to be implemented in similar fashion.
+
+A simple object to allow referencing other components in the OpenAPI document, internally and externally.
+
+The `$ref` string value contains a URI [RFC3986](https://tools.ietf.org/html/rfc3986), which identifies the location of the value being referenced.
+
+See the rules for resolving [Relative References](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#relativeReferencesURI).
+
+##### Fixed Fields
+Field Name | Type | Description
+---|:---:|---
+<a name="referenceRef"></a>$ref | `string` | **REQUIRED**. The reference identifier. This MUST be in the form of a URI, with a prefix of `#/components/inputs` or `#/components/parameters`.
+<a name="referenceValue"></a>value | `string` | The value of the Input or Parameter which by default SHOULD override that of the referenced component.

```

Comment:
This is a dramatic departure from how `$ref` is used in every other specification I've seen, and will cause problems with anyone trying to use standard tooling.  In particular, if this is done with `$ref` inside of schemas, you will not be able to use standard draft 2020-12 JSON Schema implementations, which was the main reason that OAS 3.1 adopted 2020-12 in full.

I'm kind of dropping in here without context- is there somewhere I can read up on the reason for this usage?   don't see any discussion of it in the referenced issue.

Reply:
Probably makes sense to discuss at our meeting this coming Wednesday, can you make it? Also, might be worth taking a look at the pet store example which I've marked up in this PR, that should at least convey the intention of this use of $ref. But certainly open to hearing improvements.

Reply:
I'll try to make the meeting.  My main concern here is that this proposal seems to be a bit disconnected from the general history and usage of `$ref`.  Support for `$ref` has been a challenge in the OAS tooling community, and has often been addressed by having standalone tools that understand `$ref` which are used in conjunction with OAS-specific tools.  Adding substantially different behavior to `$ref` will break many, many things.